src/lib/molecules/AdvancedSearch/index.js

```diff
@@ -82,9 +87,11 @@ function AdvancedSearch({
 
     const inputAndPopoverWidthVariable = useMemo(
         () => ({
-            '--advanced-search-width': popoverOpen ? `${openedInputWidth}vw` : closedInputWidth
+            '--advanced-search-width': (isOpenControlled ? isOpen : popoverOpen)
+                ? `${openedInputWidth}vw`
+                : closedInputWidth
         }),
-        [popoverOpen]
+        [popoverOpen, isOpenControlled, isOpen]
```

Comment:
Here you can include these `openedInputWidth` and `closedInputWidth` props. As they can be changed. It's just a suggestion think about it.